### ISSUE TITLE
Need more help for building merkaator from surce in Linux.

### ISSUE THREAD

**Issue by Foxofoxes** [openstreetmap/merkaartor#54]
The Install file needs more information.

It's really hard to build from source, even with the required development libraries.
There are a lot of issues we face and doesn't seems to be any help out there.

<b>some issues</b>:

[1] fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).

[2] Reading /home/usr/Desktop/merkaartor-0.18.2/plugins/background/MWalkingPapersBackground/MWalkingPapersBackground.pro
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.

[3] In file included from interfaces/IImageManager.cpp:20:0:
interfaces/IImageManager.h:26:33: fatal error: QNetworkAccessManager: No such file or directory
 #include <QNetworkAccessManager>
                                 ^
compilation terminated.
make: *** [IImageManager.o] Error 1

[4] In file included from common/MapView.h:4:0,
                 from Features/Feature.h:8,
                 from Features/Features.h:2,
                 from Backend/MemoryBackend.h:4,
                 from common/Global.h:19,
                 from common/Global.cpp:13:
common/Projection.h:12:22: fatal error: proj_api.h: No such file or directory
 #include <proj_api.h>
                      ^
compilation terminated.
make[1]: *** [Global.o] Error 1

[5] Makefile:843: warning: overriding commands for target `moc_MapView.cpp'
Makefile:789: warning: ignoring old commands for target `moc_MapView.cpp'

[6] Project ERROR: Unknown module(s) in QT: webkitwidgets svg

I used some other commands trying to make it properly with no luck:

qmake-qt4 -project
Even some git commands to avoid errors: git init

I'm using the following libraries:
g++ 4.8
qt 4.8.5, qt4-dev-tools, qt4-qtconfig and its dependencies and dependants.
libgeo-proj4-perl 1.05
git 1.9.1
and the other 2 software prerequisites, not exactly the ones that you recommend but they do the job. (I have Installed a previous Merkaartor version and it works great)

No matter what I do, I get lots of errors, And yes, I've followed the instructions, they are pretty easy but doesn't seem to do the job.

I downloaded the package from the merkaartor official website: merkaartor.be
I'm working on Ubuntu 14.04.03.

Almost every Merkaartor required development libraries are a fresh install, and some other libraries too, git by example, so I have nothing set up if required. Why does it need to be bundled as a git project?

Maybe I'm missing something, but I am not used to Qt or git. I could just download a built package, but there are not any latest builds for Ubuntu, I discovered some unofficial plugins for Merkaartor and I need to build Merkaartor for being able to use them.

**Comment by Krakonos** [comment]
Thanks. I've just compile the package on my system and seems to work fine, except the version is not correctly displayed (yey, I messed that one up). But let's get to your issues:

First, it would be beneficial if you'd include the commands that are causing them. I'll be guessing:

[1] Is output from git checkout or similar? The instructions in INSTALL that tell you to do a checkout are meant to be used after a git clone.

[2] Don't mind these. This is the result of running git describe to get version.

[3] Looks like some of the headers are missing, or some include stuff is wrong in Merkaartor. Try checking if the libqt4-dev is installed, I'll check the plugin.

[4] Means you're missing libproj-dev. You mention libgeo-proj4-perl, which are just perl bindings for the lib.

[5] I'm not sure about this one. Let's ignore it for now, and I'll check the automated builds if it happens there too. I guess it is part of the qmake/make magic.

[6] The webkitwidgets module is available only in Qt5, BUT should be ignored by reasonably new Qt (my qt4 deals with it). The problem is probably the svg module. Please check if you have qt4-svg installed.

TL;DR: On Ubuntu, I install the required packages with this command (used in .travis.yml for automated tests):

apt-get install libgdal-dev libproj-dev libqt4-xml libqt4-network libqt4-gui libqt4-svg libqt4-webkit libqt4-dev qt4-qmake

Could you point me to these plugins you want to use? I'd like to take a look and probably link them on the site if they're useful (and provide some support installing them).

**Comment by Foxofoxes** [comment]
<b>[1]</b> When using the default commands:

`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake -r`
`Reading /home/jdg/Desktop/merkaartor-0.18.2/src/src.pro
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Project ERROR: Unknown module(s) in QT: webkitwidgets svg
jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ make
cd src/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/jdg/Desktop/merkaartor-0.18.2/src/src.pro -o Makefile ) && make -f Makefile 
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Project ERROR: Unknown module(s) in QT: webkitwidgets svg
make: *** [sub-src-make_first] Error 3`

<b>[2]</b> Tried to solve the issue by myself doing an Internet research:

`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ git init`
`Initialised empty Git repository in /home/jdg/Desktop/merkaartor-0.18.2/.git/`

But keep seeing the following errors:

`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake -r`
`Reading /home/jdg/Desktop/merkaartor-0.18.2/src/src.pro
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
Project ERROR: Unknown module(s) in QT: webkitwidgets svg
jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ make
cd src/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/jdg/Desktop/merkaartor-0.18.2/src/src.pro -o Makefile ) && make -f Makefile 
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
Project ERROR: Unknown module(s) in QT: webkitwidgets svg
make: *** [sub-src-make_first] Error 3`

<b>[3]</b> Tried handling the sources as a project:
I don't really need to use the qmake-qt4 command, my system uses qt4 as default.

`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake-qt4 -project`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake -r`
`WARNING: TARGET is empty`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ make`
`Makefile:2163: warning: overriding commands for target `moc_MapView.cpp'
Makefile:1467: warning: ignoring old commands for target `moc_MapView.cpp'
Makefile:11804: warning: overriding commands for target `main.moc'
Makefile:11670: warning: ignoring old commands for target `main.moc'
Makefile:11924: warning: overriding commands for target `main.moc'
Makefile:11804: warning: ignoring old commands for target `main.moc'
Makefile:17011: warning: overriding commands for target `MapView.o'
Makefile:13182: warning: ignoring old commands for target `MapView.o'
Makefile:45951: warning: overriding commands for target `mapadapter.o'
Makefile:35464: warning: ignoring old commands for target `mapadapter.o'
Makefile:47775: warning: overriding commands for target `main.o'
Makefile:13023: warning: ignoring old commands for target `main.o'
Makefile:47909: warning: overriding commands for target `main.o'
Makefile:47775: warning: ignoring old commands for target `main.o'
Makefile:48029: warning: overriding commands for target `main.o'
Makefile:47909: warning: ignoring old commands for target `main.o'
Makefile:48449: warning: overriding commands for target `main.o'
Makefile:48029: warning: ignoring old commands for target `main.o'
make: Nothing to be done for `first'.`

<b>[4]</b> Using qmake makefile option:

`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake-qt4 -makefile`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ qmake -r`
`WARNING: TARGET is empty`
`jdg@jdg-Inspiron1545:~/Desktop/merkaartor-0.18.2$ make`
`Makefile:843: warning: overriding commands for target `moc_MapView.cpp'
Makefile:789: warning: ignoring old commands for target `moc_MapView.cpp'
Makefile:2065: warning: overriding commands for target `main.moc'
Makefile:2061: warning: ignoring old commands for target `main.moc'
Makefile:2069: warning: overriding commands for target `main.moc'
Makefile:2065: warning: ignoring old commands for target `main.moc'
Makefile:2903: warning: overriding commands for target `MapView.o'
Makefile:2413: warning: ignoring old commands for target `MapView.o'
Makefile:5981: warning: overriding commands for target `mapadapter.o'
Makefile:5214: warning: ignoring old commands for target `mapadapter.o'
Makefile:6074: warning: overriding commands for target `main.o'
Makefile:2396: warning: ignoring old commands for target `main.o'
Makefile:6078: warning: overriding commands for target `main.o'
Makefile:6074: warning: ignoring old commands for target `main.o'
Makefile:6082: warning: overriding commands for target `main.o'
Makefile:6078: warning: ignoring old commands for target `main.o'
Makefile:6101: warning: overriding commands for target `main.o'
Makefile:6082: warning: ignoring old commands for target `main.o'
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/MainWindow.ui -o ui_MainWindow.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic mobilemerk/src/MobileMainWindow.ui -o ui_MobileMainWindow.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/AboutDialog.ui -o ui_AboutDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/DownloadMapDialog.ui -o ui_DownloadMapDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/GotoDialog.ui -o ui_GotoDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/MultiProperties.ui -o ui_MultiProperties.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/PropertiesDialog.ui -o ui_PropertiesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/RelationProperties.ui -o ui_RelationProperties.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/TerraceDialog.ui -o ui_TerraceDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/common/UploadMapDialog.ui -o ui_UploadMapDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/DirtyDock.ui -o ui_DirtyDock.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/FeaturesDock.ui -o ui_FeaturesDock.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/MinimumRelationProperties.ui -o ui_MinimumRelationProperties.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/MinimumRoadProperties.ui -o ui_MinimumRoadProperties.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/MinimumTrackPointProperties.ui -o ui_MinimumTrackPointProperties.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/PhotoLoadErrorDialog.ui -o ui_PhotoLoadErrorDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Docks/StyleDock.ui -o ui_StyleDock.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/GPS/qgpsmainwindowui.ui -o ui_qgpsmainwindowui.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/ImportExport/ExportDialog.ui -o ui_ExportDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/ImportExport/ImportCSVDialog.ui -o ui_ImportCSVDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Interactions/CreateDoubleWayDock.ui -o ui_CreateDoubleWayDock.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Interactions/CreateRoundaboutDock.ui -o ui_CreateRoundaboutDock.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Layers/FilterEditDialog.ui -o ui_FilterEditDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Layers/LayerWidget.ui -o ui_LayerWidget.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Layers/LicenseDisplayDialog.ui -o ui_LicenseDisplayDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/NameFinder/namefinderwidget.ui -o ui_namefinderwidget.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/PaintStyle/PaintStyleEditor.ui -o ui_PaintStyleEditor.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/FilterPreferencesDialog.ui -o ui_FilterPreferencesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/OsmServerWidget.ui -o ui_OsmServerWidget.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/PreferencesDialog.ui -o ui_PreferencesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/ProjPreferencesDialog.ui -o ui_ProjPreferencesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/TMSPreferencesDialog.ui -o ui_TMSPreferencesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Preferences/WMSPreferencesDialog.ui -o ui_WMSPreferencesDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/QToolBarDialog/qttoolbardialog.ui -o ui_qttoolbardialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Render/NativeRenderDialog.ui -o ui_NativeRenderDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Sync/SyncListDialog.ui -o ui_SyncListDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Utils/PictureViewerDialog.ui -o ui_PictureViewerDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Utils/ProjectionChooser.ui -o ui_ProjectionChooser.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Utils/SelectionDialog.ui -o ui_SelectionDialog.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic src/Utils/TagSelectorWidget.ui -o ui_TagSelectorWidget.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic plugins/background/MCadastreFranceBackground/qadastre/cadastrebrowser.ui -o ui_cadastrebrowser.h
/usr/lib/x86_64-linux-gnu/qt4/bin/uic plugins/background/MCadastreFranceBackground/qadastre/searchdialog.ui -o ui_searchdialog.h
g++ -c -m64 -pipe -O2 -Wall -W -D_REENTRANT -DQT_NO_DEBUG -DQT_GUI_LIB -DQT_CORE_LIB -DQT_SHARED -I/usr/share/qt4/mkspecs/linux-g++-64 -I. -I/usr/include/qt4/QtCore -I/usr/include/qt4/QtGui -I/usr/include/qt4 -I. -Iinterfaces -Isrc/Preferences -Isrc/common -Isrc/Backend -Isrc/Features -Imobilemerk/src -Isrc/PaintStyle -Isrc/Utils -Isrc/Layers -Isrc/Docks -I3rdparty/qtsingleapplication-2.6_1-opensource/src -Isrc -Isrc/Commands -Isrc/ImportExport -Isrc/Interactions -Isrc/Sync -Isrc/QMapControl -Isrc/Tools -Isrc/Render -Isrc/GPS -Isrc/QToolBarDialog -Imobilemerk/thirdparty/MouseMachine -Iplugins/background/MGosmoreBackground -Iplugins/background/MNavitBackground -Iinclude -Isrc/NameFinder -Isrc/TagTemplate -Isrc/qextserialport -I3rdparty/quazip-0.7/quazip -I3rdparty/quazip-0.7/qztest -Iplugins/background/MCadastreFranceBackground -Iplugins/background/MCadastreFranceBackground/qadastre -Iplugins/background/MGdalBackground -Iplugins/background/MGeoTiffBackground -Iplugins/background/MMsBingMapBackground -Iplugins/background/MSpatialiteBackground -Iplugins/background/MWalkingPapersBackground -Iplugins/background/MYahooBackground -Iplugins/background/MYahooTiledBackground -Isrc/Tools/QFatFs -I. -I. -o IImageManager.o interfaces/IImageManager.cpp
In file included from interfaces/IImageManager.cpp:20:0:
interfaces/IImageManager.h:26:33: fatal error: QNetworkAccessManager: No such file or directory
 #include <QNetworkAccessManager>
                                 ^
compilation terminated.
make: *** [IImageManager.o] Error 1`

Did some other things, reinstalled packages, changed packages, searched for more results (related to Qt and git errors), checked if there were any broken packages and many other things that I don't remember right now. I wasn't able to find something helpful and related to merkaator on the Internet.

- <b>libqt4-dev</b> it's installed (4.8.5) (Actually 4:4.8.5+git192-g085f851+dfsg-2ubuntu4.1)

- <b>libproj-dev</b> it's also installed (1.05-1)

- And yes, <b>libqt4-svg</b> it's installed (4.8.5) (4:4.8.5+git192-g085f851+dfsg-2ubuntu4.1)
All these packages are the default ones, provided by the Ubuntu developers.

Ran the command you posted above, I was missing libqt4-webkit libqt4-gui, then tried again but the error [1] appeared.
Then I discovered Ubuntu (10 and forward versions) 14.04 is missing libqt4-webkit-dev, uses libqtwebkit-dev (2.3.2-0ubuntu7) instead.

<b>Installed qt5:</b>
apt-get install qt5-qmake qt5-default libqt5webkit5-dev libqt5svg5-dev

> Am I missing something?

<b>Ran the following commands:</b>
`/usr/lib/x86_64-linux-gnu/qt5/bin/qmake`
`/usr/lib/x86_64-linux-gnu/qt5/bin/qmake -r`
`make`

Note: running "qmake-qt5" won't work, even when it's supposed to work, maybe it's a different command.

Doing so I was able to compile merkaator.

<b>[5]</b> These are the plugins, I am not really sure if they work as they should. I thought these were yours.
https://code.launchpad.net/~koying/merkaartor/unofficial-plugins

Well, thank you. I can't believe it was that easy, but the good thing it's I possibly added support for compiling on Ubuntu.

Yiff yiff

**Comment by triumphov** [comment]
Good afternoon. I have a mistake :
/home/triumphov/merkaartor/src/common/Projection.h:12: error: proj_api.h: No such file or directory
  #include <proj_api.h>
I would be very grateful if you could help me compile the project.

**Comment by Krakonos** [comment]
Hi! It seems like you don't have proj4 installed. Please see COMPILE.md for dependencies and instructions on how to install them on your system. Please note that if you have some debian derivative, you need to have the appropriate -dev package installed as well.

Also, please don't reuse old issues, closed two years ago. Feel free to create a new one, even if it's a minor problem or a question.
